core/controller/src/main/scala/org/apache/openwhisk/core/loadBalancer/ShardingContainerPoolBalancer.scala

```diff
@@ -244,23 +294,33 @@ class ShardingContainerPoolBalancer(
 
     val isBlackboxInvocation = action.exec.pull
     val actionType = if (!isBlackboxInvocation) "managed" else "blackbox"
-
     val (invokersToUse, stepSizes) =
       if (!isBlackboxInvocation) (schedulingState.managedInvokers, schedulingState.managedStepSizes)
       else (schedulingState.blackboxInvokers, schedulingState.blackboxStepSizes)
     val chosen = if (invokersToUse.nonEmpty) {
       val hash = ShardingContainerPoolBalancer.generateHash(msg.user.namespace.name, action.fullyQualifiedName(false))
       val homeInvoker = hash % invokersToUse.size
       val stepSize = stepSizes(hash % stepSizes.size)
-      val invoker = ShardingContainerPoolBalancer.schedule(
+      val invoker: Option[(InvokerInstanceId, Boolean)] = ShardingContainerPoolBalancer.schedule(
         action.limits.concurrency.maxConcurrent,
         action.fullyQualifiedName(true),
         invokersToUse,
         schedulingState.invokerSlots,
         action.limits.memory.megabytes,
         homeInvoker,
         stepSize)
-      invoker
+      invoker.map {
+        case (invoker, isOverloaded) => {
+          if (isOverloaded) {
+            if (isBlackboxInvocation) {
+              MetricEmitter.emitCounterMetric(LoggingMarkers.BLACKBOX_SYSTEM_OVERLOAD)
+            } else {
+              MetricEmitter.emitCounterMetric(LoggingMarkers.MANAGED_SYSTEM_OVERLOAD)
+            }
+          }
+          invoker
+        }
+      }
```

Comment:
As you're only executing a side-effect on the option, please use `foreach` and just return the outer invoker value, like this:

```scala
invoker.foreach {
  case (_, true) =>
    val metric = if(isBlackboxInvocation) LoggingMarkers.BLACKBOX_SYSTEM_OVERLOAD else LoggingMarkers.MANAGED_SYSTEM_OVERLOAD
    MetricEmitter.emitCounterMetric(metric)
  case _ =>
}
invoker.map(_._1)
```

You don't care about the return values of the `emitCounterMetric`, hence they must produce a side-effect. If not for any performance concerns (which is not the case here), one should always use `foreach` to produce those, because `foreach` returns Unit, signifying the side-effect.